### ISSUE TITLE
Remove view number from Macro blocks

### ISSUE THREAD

**Issue by brunoffranca** [nimiq/core-rs-albatross#189]
This was supposed to be done in issue #186, but wasn't because the view number was used in too many places and the other changes were more urgent. Also, it is a good idea to merge the change to Tendermint before.
This is not strictly necessary, but it's a nice improvement since in theory we don't need the view number.

**Comment by brunoffranca** [comment]
Turns out that we need the round/view number also in the macro blocks.

**Comment by brunoffranca** [comment]
After some discussion, it was decided that the round/view number for the macro blocks should be in the justification and not in the header. So this is still necessary.

**Comment by brunoffranca** [comment]
Turns out that Macro blocks do need to have a view number, and that it is different from the round number used in Tendermint.

What happens is that the Macro block proposals (or headers) do contain one piece of information that is specific to the block proposer: the VRF seed. In order to verify a macro block we need to know who created the proposal, to be able to check the seed.

Also the view number can be different from the round number used in Tendermint. In Tendermint there are situations when a proposer does not create a proposal itself, but instead reuses a proposal that it received before. In effect, the round number can increase without the view number increasing.

TLDR: We need to have both view numbers and round numbers in the Macro blocks.
